scripts/schemas/board-schema.yml

```diff
@@ -78,3 +78,56 @@ mapping:
     type: seq
     sequence:
       - include: board-schema
+  run_once:
+    type: seq
+    desc: |
+          Allows for restricting west flash commands when using sysbuild to run once per given
+          grouping of boards. This is to allow for future image program cycles to not erase the
+          flash of a device which has just been programmed by another image.
+    sequence:
+      - type: map
+        mapping:
+          regex;(.*):
+            required: true
+            type: seq
+            desc: |
+                  A list of commands which should be limited to running once per invocation of
+                  west flash for a set of boards
+            sequence:
+              - type: map
+                mapping:
+                  boards:
+                    required: true
+                    type: seq
+                    desc: |
+                          A grouping of boards which the command should apply to. Can be used
+                          multiple times to have multiple groups.
```

Comment:
what does a group mean in this context ?

Why not just a list of boards ?
What happens if I move a board from one group to another, does it change anything ?

Reply:
It changes a lot, see https://github.com/zephyrproject-rtos/zephyr/pull/69748#discussion_r1517464462